git-refresher/moving-between-branches.qmd

```diff
@@ -1,18 +1,61 @@
---- 
+---
 title: "Moving between branches"
 ---
 
-...
+This section will guide you through switching branches in your repo.
+
+Switching branches is useful for when we might need to pause our work on one branch and continue working on an issue in another branch. Before you switch branches it is important to commit and push the changes in the branch you're currently in. It's also important to pull changes after you switch to your new branch.
 
 ::: panel-tabset
 ## Git Bash
 
+First you need to update your local repo with the changes made to the remote one. This ensures that you have a full list of remote branches and the latest changes in the code to avoid conflicts. Enter the command below:
+
+`git pull`
+
+In Git Bash, you can switch branches using either of the following options:
+
+`git checkout`
+
+OR
+
+`git switch`
+
 ## GitHub Desktop
 
 ## VS Code
 
 ## RStudio
 
+-   Go to the right hand side of RStudio and click on the Git tab.
+
+-   Click on the downwards arrow to "pull" the latest changes from the remote repo. This ensures that:
+
+    -   you have a full list of remote branches.
+    -   you have the latest changes in the code to avoid conflicts.
+
+-   Click on the drop down menu of branches. Usually "main" is the default branch shown before switching.
+
+-   Click on `<github_id>_main`
+
+    -   The example in the screenshot below show a branch name specific to another user. You need to click on your own branch.
+
+![](images/r_switch_branch.png){fig-alt="The Git panel in RStudio with red rectangles highlighting the downwards pull arrow, the branches drop down menu and an example of a branch we want to switch to."}
+
 ## PyCharm
 
+-   Click on the branches drop down menu at the top of the screen.
+
+-   Click the Fetch icon to "pull" the latest changes from the remote repo. This ensures that:
+
+    -   you have a full list of remote branches.
+    -   you have the latest changes in the code to avoid conflicts.
+
+-   Click on "Remote" to see the full list of remote branches if you can't already.
+
+-   Click on `<github_id>_main` and then click "Checkout".
```

Comment:
As above, I think assume at this stage that they'll be progressing through linearly so will be on their own branch already.